robot_modeling/DQ_Kinematics.m

```diff
@@ -367,6 +369,29 @@ function set_base_frame(obj, base_frame)
             J = 2*vec4(robot_line - workspace_line)'*Jl;
             
         end
+
+
+        function residual = line_to_line_angle_residual(robot_line, ...
+                 workspace_line, workspace_line_derivative) 
+        % LINE_TO_LINE_ANGLE_RESIDUAL(robot_point, workspace_line,
+        % workspace_line_derivative) returns the residual related to the
+        % moving line orientation in the workspace that is independent of 
+        % the robot motion (i.e., which does not depend on the robot joint
+        % velocities). 
+        %
+        % For more details see Eq. 4.30 of Juan José Quiroz-Omaña."Whole-Body 
+        % Control of Humanoids Robots at Second Order Kinematics Under 
+        % Unilateral Constraints." PhD thesis, Federal University of
+        % Minas Gerais, 2021.
+        % https://repositorio.ufmg.br/handle/1843/38700
+
+                if ~is_line(robot_line) || ~is_line(workspace_line)
+                    error(['robot_line and'...
+                           'workspace_line must be lines.']);
+                end
+                residual = 2*dot(robot_line-workspace_line,...
+                            -1.0*workspace_line_derivative);
```

Comment:
It is not necessary to explicitly multiply by -1. The negative sign is a unary operator. So, it suffices to use `-workspace_line_derivative`.

Reply:
@bvadorno Thank you for your feedback. I implemented the modifications you suggested. Please let me know if something else is required.

Best regards, 

Juancho